optimistic_lock.hpp

```diff
@@ -199,15 +212,41 @@ class [[nodiscard]] optimistic_lock final {
       return *this;
     }
 
+    // Point to the same lock and have the same version for that lock.
+    bool operator==(const read_critical_section &other) const noexcept {
+      return lock == other.lock && version == other.version;
+    }
+
+    // Unlock iff it is not yet unlocked.
+    //
+    // Note: In a DEBUG build, this clears the [lock] pointer,
+    // rendering the RCS unusable, and decrements the read_lock_count.
```

Comment:
The RCS should be unusable in all builds, even if only debug build asserts/crashes

Reply:
I am not sure of the clarification here.  Do you mean that the RCS is conceptually invalid after unlock()?  Yes.  But if I parse the double negative correctly the lock pointer is only cleared to nullptr if this is not a debug build.  

```
#ifndef NDEBUG
      lock = nullptr;
#endif
```

I will clarify the comment that unlock() means that the RCS is invalid.  But there is also the gotcha behavior with the lock pointer clearing to nullptr that I found worth surfacing in the method contract.

Reply:
Updated the documentation and resolving the thread

Reply:
I read the original comment as "something bad will happen on the debug build [but fine on release build]", and did not want to imply the implied release build part. Maybe it's just me, but in any case thanks for addressing the potential ambiguity